src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -1036,30 +952,44 @@ private void ParseHelper(string[] args)
                 }
             }
 
-            if (_showHelp)
-            {
-                ShowHelp();
-            }
-
-            if (_showBanner && !_showHelp)
-            {
-                DisplayBanner();
-
-                if (UpdatesNotification.CanNotifyUpdates)
-                {
-                    UpdatesNotification.ShowUpdateNotification(_hostUI);
-                }
-            }
-
             Dbg.Assert(
                     ((_exitCode == ConsoleHost.ExitCodeBadCommandLineParameter) && _abortStartup)
                 || (_exitCode == ConsoleHost.ExitCodeSuccess),
                 "if exit code is failure, then abortstartup should be true");
         }
 
-        private void WriteCommandLineError(string msg, bool showHelp = false, bool showBanner = false)
+#if DEBUG
+        private void WaitingRemoteDebugger(PSHostUserInterface hostUI)
+        {
+            // this option is useful when debugging ConsoleHost remotely using VS remote debugging, as you can only
+            // attach to an already running process with that debugger.
+            if (_wait)
+            {
+                // This does not need to be localized: its chk only
+                ((ConsoleHostUserInterface)hostUI).WriteToConsole("Waiting - type enter to continue:", false);
+                hostUI.ReadLine();
+            }
+        }
+#endif
+
+        internal void ShowErrorHelpBanner(PSHostUserInterface hostUI, string? bannerText, string? helpText, string[] args)
```

Comment:
The parameter `args` can be removed, right?

Reply:
Removed.

(Initial intention was only merge early parsing with main parsing without any code modifications but I found some minor bugs, Jim ask me to add tests... and we fall in refactoring :-) I guess there are more places for fixes.)

Reply:
Then place review your own changes and fix those places. I don't have the time to find all issues like this during the review.